demo/src/App.tsx

```diff
@@ -35,15 +40,26 @@ export default function App() {
 
     return (
         <main className="container">
-            <Header />
+            <Header isDarkMode={isDarkMode} toggleTheme={toggleTheme} />
             {failureBanner && (
                 <Alert severity="error" data-testid="json-parse-alert">
                     {failureBanner}
                 </Alert>
             )}
             <Stack direction="row" spacing={2}>
                 <Box sx={{ width: "40%" }}>
-                    <h2>Input</h2>
+                    <Typography
+                        variant="h2"
+                        color="textPrimary"
+                        gutterBottom
+                        sx={{
```

Comment:
Overall, I'm not a fan of including styles like this in the components. Globally, do not use `sx`, and instead update the CSS file.